akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -328,7 +328,9 @@ private[akka] object PromiseActorRef {
   private case object Stopped
   private final case class StoppedWithPath(path: ActorPath)
 
-  def apply(provider: ActorRefProvider, timeout: Timeout, targetName: String, message: Any, sender: ActorRef = Actor.noSender): PromiseActorRef = {
+  private val ActorStopResult = Failure(new ActorKilledException("Stopped"))
+
+  def apply(provider: ActorRefProvider, timeout: Timeout, targetName: ⇒ String, message: Any, sender: ActorRef = Actor.noSender): PromiseActorRef = {
```

Comment:
Wouldn't it be more appropriate to change the signature to: "target: Any" and then do "target.toString" when/if needed? (you're going to have to close over it anyway in the Function0